src/main/java/org/folio/consortia/service/impl/SharingRoleService.java

```diff
@@ -81,7 +96,49 @@ protected void validateSharingConfigRequestOrThrow(UUID roleId, SharingRoleReque
 
   @Override
   protected Set<String> findTenantsForConfig(SharingRoleRequest request) {
-    return sharingRoleRepository.findTenantsByRoleId(request.getRoleId());
+    return sharingRoleRepository.findTenantsByRoleName(request.getRoleName());
+  }
+
+  @Override
+  protected void syncConfigWithTenants(Set<String> sharedConfigTenants, SharingRoleRequest request) {
+    String roleName = request.getRoleName();
+    String centralTenantId = folioExecutionContext.getTenantId();
+    log.debug("syncConfig:: Trying to syncing sharing role table with roles table for role '{}' and tenant '{}'",
+      request.getRoleName(), centralTenantId);
+
+    if (sharingRoleRepository.existsByRoleNameAndTenantId(roleName, centralTenantId)) {
+      log.info("syncConfig:: Role '{}' with central tenant '{}' already exists, Syncing with other tenants: {}",
+        request.getRoleName(), centralTenantId, sharedConfigTenants);
+
+      sharedConfigTenants.stream()
+        .filter(tenantId -> !tenantId.equals(centralTenantId))
+        .forEach(memberTenantId -> syncSharingRoleWithRoleInTenant(request, roleName, memberTenantId));
+      return;
+    }
+
+    log.info("syncConfig:: Role '{}' not found, trying to sync with only central tenant '{}'" +
+        " because role haven't shared with other tenants yet", request.getRoleId(), centralTenantId);
+    syncSharingRoleWithRoleInTenant(request, roleName, centralTenantId);
+  }
+
+  private void syncSharingRoleWithRoleInTenant(SharingRoleRequest request, String roleName, String tenantId) {
+    try (var ignored = new FolioExecutionContextSetter(prepareContextForTenant(tenantId, moduleMetadata, folioExecutionContext))) {
```

Comment:
some logic can be moved in base class, for example you can remain here saving of request in DB and flow logic move

Reply:
setting don't need sync and policy don't need member tenant sync. It can be complex. I will try, if possible